torch/utils/checkpoint.py

```diff
@@ -166,9 +166,11 @@ def checkpoint(function, *args, **kwargs):
     be considered as part of autograd.
 
     .. warning::
-        Checkpointing currently only supports :func:`torch.autograd.backward`
-        and only if its `inputs` argument is not passed. :func:`torch.autograd.grad`
-        is not supported.
+        If ``use_reentrant=True`` is specified, checkpointing currently only
```

Comment:
I feel like we could re-order some of the warnings. Especially because most of them only apply to the use_reentrant=True case.

Reply:
Moved this warning to the bottom then, and still kept the wording as use_reentrant=True is the default for now. Let me know if they could be structured better.